### PR TITLE
Fix return value using Yao::OldSample.list

### PR DESCRIPTION
**Pull request by hsbt**
Fix return values using `Yao::OldSample.list`. You got always same results with same query and different meter.

### CODE REVIEW

File: lib/yao/resources/old_sample.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def user
 
     # get /v2/meters/{id} returns samples!
     def self.list(meter_name, query={})
-      cache_key = query.values.join
+      cache_key = [meter_name, *query].join
```
Comment (udzura) [inline lib/yao/resources/old_sample.rb]: `query` is a Hash, would you mean `*query` to apply `Hash#to_a` ?

Comment (hsbt) [inline lib/yao/resources/old_sample.rb]:
```rb
>> [*{foo: :bar, buzz: :foo}]
=> [[:foo, :bar], [:buzz, :foo]]
>> [{foo: :bar, buzz: :foo}.to_a]
=> [[[:foo, :bar], [:buzz, :foo]]]
```

it's tiny difference behavior.

Comment (udzura) [inline lib/yao/resources/old_sample.rb]: Oh yeah, understood.

**Comment by hsbt** [comment]
Thank you for your review.

**Pull request merged**